crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -147,15 +149,24 @@ impl TransactionBlock {
     ) -> Result<Connection<String, Event>> {
         let page = Page::from_params(ctx.data_unchecked(), first, after, last, before)?;
         let mut connection = Connection::new(false, false);
-        let Some((prev, next, cs)) = page.paginate_indices(self.stored.events.len()) else {
+        let len = match &self.tx_data {
+            Either::Left(stored) => stored.events.len(),
+            Either::Right(executed) => executed.events.len(),
+        };
+        let Some((prev, next, cs)) = page.paginate_indices(len) else {
             return Ok(connection);
         };
 
         connection.has_previous_page = prev;
         connection.has_next_page = next;
 
         for c in cs {
-            let event = Event::try_from_stored_transaction(&self.stored, *c).extend()?;
+            let event = match self.tx_data {
+                Either::Left(ref stored) => {
+                    Event::try_from_stored_transaction(stored, *c).extend()?
+                }
+                Either::Right(ref executed) => executed.events[*c].clone(),
```

Comment:
And then here we would construct an Event where stored is empty and native is the element at c

Reply:
yup